dace/transformation/dataflow/otf_map_fusion.py

```diff
@@ -139,22 +144,69 @@ def apply(self, graph: SDFGState, sdfg: SDFG):
             new_param = param + "_local"
             advanced_replace(subgraph, param, new_param)
 
-        # Phase 1: Re-connect access nodes
-        # a. Collect out access nodes of first map
-        intermediate_dnodes = set()
-        for edge in graph.out_edges(self.first_map_exit):
-            node = edge.dst
-            if not isinstance(node, nds.AccessNode):
+        # Add local buffers for array-like OTFs
+        for edge in graph.out_edges(self.second_map_entry):
+            if edge.data is None or edge.data.data != self.array.data:
                 continue
 
-            intermediate_dnodes.add(node)
+            xform = InLocalStorage()
+            xform._sdfg = sdfg
+            xform.state_id = sdfg.node_id(graph)
+            xform.node_a = edge.src
+            xform.node_b = edge.dst
+            xform.array = self.array.data
+            if xform.can_be_applied(graph, expr_index=0, sdfg=sdfg):
+                InLocalStorage.apply_to(sdfg=sdfg,
+                                        node_a=edge.src,
+                                        node_b=edge.dst,
+                                        options={"array": self.array.data},
+                                        verify=True,
```

Comment:
verify=True would just run can_be_applied again, so you can safely set it to False